common/ferrostar/src/algorithms.rs

```diff
@@ -379,9 +357,9 @@ fn distance_along(point: &Point, linestring: &LineString) -> Option<f64> {
 
             // Compute distance to the line (sadly Euclidean only; no haversine_distance in GeoRust
             // but this is probably OK for now)
-            let segment_distance_to_point = segment.euclidean_distance(point);
+            let segment_distance_to_point = Euclidean::distance(&segment, point);
             // Compute total segment length in meters
-            let segment_length = segment_linestring.haversine_length();
+            let segment_length = segment_linestring.length::<Haversine>();
```

Comment:
For example, you could double down on the more accurate but slower `segment_linestring.length::<Geodesic>()`

Reply:
Yeah, exactly. That's probably worth evaluating properly. I've opened #329 to follow up on this. This is _probably_ a case where Geodesics could make sense. On the other hand, I am pretty sure Valhalla uses haversine exclusively and is generally fine 😂